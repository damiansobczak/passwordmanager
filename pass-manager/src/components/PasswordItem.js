import React from "react";
import "./PasswordItem.scss";
class PasswordItem extends React.Component {
  showPassword = id => {
    const password = document.getElementById(`${id}`);
    if (password.type === "password") {
      password.type = "text";
    } else {
      password.type = "password";
    }
  };

  copyPassword = id => {
    let textArea = document.createElement("textarea");
    const password = document.getElementById(`${id}`);
    textArea.value = password.value;
    textArea.style.opacity = "0";
    document.body.appendChild(textArea);

    const copy = new Promise((resolve, reject) => {
      textArea.select();
      const copied = document.execCommand("copy");
      copied === true ? resolve(textArea) : reject(new Error("Nie skopiowano! Coś poszło nie tak."));
    });

    copy
      .then(password => {
        textArea.remove();
        alert("Hasło zostało skopiowane");
      })
      .catch(new Error("Ops! Something went wrong!"));
  };

  render() {
    return (
      <div className="password">
        <div className="password-name">Google.com</div>
        <input type="password" value="Twójstary" className="password-hash" id="password-id-1" />
        <div className="password-date">12-03-2019</div>
        <div className="password-actions">
          <div className="white red-text waves-effect waves-teal btn-flat password-reveal" onClick={() => this.showPassword("password-id-1")}>
            Odsłoń
          </div>
          <a className="blue accent-4 accent-bluewaves-effect waves-light btn password-copy" onClick={() => this.copyPassword("password-id-1")}>
            kopiuj
          </a>
        </div>
      </div>
    );
  }
}

export default PasswordItem;
