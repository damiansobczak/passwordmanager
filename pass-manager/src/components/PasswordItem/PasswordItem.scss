.password {
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-bottom: 1px solid rgba(160, 160, 160, 0.2);
  padding: 12px 0;

  &:last-child {
    border-bottom: none;
  }

  input.password-hash,
  input.password-hash:focus {
    border: none;
    padding: 0;
    margin: 0;
    width: auto;
    height: auto;
    box-shadow: none;
  }
}
