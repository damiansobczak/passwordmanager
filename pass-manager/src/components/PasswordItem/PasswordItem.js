import React, { Component } from "react";
import "./PasswordItem.scss";
import PropTypes from "prop-types";

class PasswordItem extends Component {
  passwordHash = React.createRef();

  showPassword = () => {
    const password = this.passwordHash.current;
    password.type === "password" ? (password.type = "text") : (password.type = "password");
  };

  copyPassword = () => {
    const password = this.passwordHash.current.value;
    let textArea = document.createElement("textarea");
    textArea.value = password;
    textArea.style.opacity = "0";
    document.body.appendChild(textArea);
    new Promise((resolve, reject) => {
      textArea.select();
      const copied = document.execCommand("copy");
      copied === true ? resolve(textArea) : reject(new Error("Nie skopiowano! Coś poszło nie tak."));
    })
      .then(password => {
        textArea.remove();
        alert("Hasło zostało skopiowane");
      })
      .catch(new Error("Ops! Something went wrong!"));
  };

  render() {
    return (
      <div className="password">
        <div className="password-name">{this.props.name}</div>
        <input type="password" readOnly defaultValue={this.props.hash} className="password-hash" ref={this.passwordHash} />
        <div className="password-date">{this.props.date}</div>
        <div className="password-actions">
          <div className="white red-text waves-effect waves-teal btn-flat password-reveal" onClick={() => this.showPassword()}>
            Odsłoń
          </div>
          <div className="blue accent-4 accent-bluewaves-effect waves-light btn password-copy" onClick={() => this.copyPassword()}>
            Kopiuj
          </div>
        </div>
      </div>
    );
  }
}

PasswordItem.propTypes = {
  name: PropTypes.string.isRequired,
  hash: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired
};

export default PasswordItem;
