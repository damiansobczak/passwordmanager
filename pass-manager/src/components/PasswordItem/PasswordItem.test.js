import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import PasswordItem from "./PasswordItem";

describe("<PasswordItem />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<PasswordItem name="Google" date="1-04-2019" key="1" hash="hashedpassword" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders password title", () => {
    const wrapper = shallow(<PasswordItem name="Google" date="" key="" hash="" />);
    expect(wrapper.find(".password-name").text()).toContain("Google");
  });

  it("renders hash", () => {
    const wrapper = shallow(<PasswordItem name="" date="" key="" hash="hashedpassword" />);
    expect(wrapper.find(".password-hash").props().defaultValue).toBe("hashedpassword");
  });

  it("renders hidden hashed password", () => {
    const wrapper = shallow(<PasswordItem name="" date="" key="" hash="hashedpassword" />);
    expect(wrapper.find(".password-hash").props().type).toBe("password");
  });

  it("renders date", () => {
    const date = new Date();
    const currentDate = `${date.getDate()}-${date.getMonth() + 1}-${date.getFullYear()}`;
    const wrapper = shallow(<PasswordItem name="" date={currentDate} key="" hash="" />);
    expect(wrapper.find(".password-date").text()).toContain(currentDate);
  });
});
