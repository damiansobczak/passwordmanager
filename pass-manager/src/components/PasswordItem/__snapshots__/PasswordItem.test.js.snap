// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordItem /> renders correctly 1`] = `
<div
  className="password"
>
  <div
    className="password-name"
  >
    Google
  </div>
  <input
    className="password-hash"
    defaultValue="hashedpassword"
    readOnly={true}
    type="password"
  />
  <div
    className="password-date"
  >
    1-04-2019
  </div>
  <div
    className="password-actions"
  >
    <div
      className="white red-text waves-effect waves-teal btn-flat password-reveal"
      onClick={[Function]}
    >
      Odsłoń
    </div>
    <div
      className="blue accent-4 accent-bluewaves-effect waves-light btn password-copy"
      onClick={[Function]}
    >
      Kopiuj
    </div>
  </div>
</div>
`;
